### DISCUSSION TITLE
How to create 1 admin on instance per server

### DISCUSSION THREAD

**Discussion by lixelv** [pocketbase/pocketbase#5383] [Q&A] [answered]
I'm building 1 app, and sometimes I need to use admin requests, like for example. I need to check, does user exists, then he is resetting password or signing up. Also checking user profiles and etc.

The question is how to make 1 admin instance per server, instead of a lot of admin authentications. Every time then user sends me request I'm authenticating in admin account, and this is inefficient.

How to make 1 admin instance per whole server, then It was created?

Note: I'm using sveltekit

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not entirely sure if that is what you are asking but you can safely create the admin SDK instance in some js file and then import it. The imported file is evaluated only once for the entire JS process no matter in how many different files you are importing it.

For example:

```js
// src/pbadmin.js
import PocketBase from "pocketbase"

const pb = new PocketBase("https://example.com");

pb.autoCancellation(false); // we need to be able to process async requests from different users

// the credentials can be filled via env variables
await pb.admins.authWithPassword(ADMIN_EMAIL, ADMIN_PASS, {
  // This will trigger auto refresh or auto reauthentication in case
  // the token has expired or is going to expire in the next 30 minutes.
  autoRefreshThreshold: 30 * 60
})

export default pb;
```

In some other file:

```js
import pbadmin from "./src/pbadmin.js"

async function someServerAction(req, resp) {
  await pbadmin.collection("example").create({ ... })
  
  ...
}
```

Since you are using SvelteKit please make sure to also check https://github.com/pocketbase/pocketbase/discussions/5313.

**Reply by lixelv** [reply]
Thank you so much, it worked! I also want to say that pocketbase is awesome! And I thought it would be great to add a new type of columns to pocketbase, generated columns. You put a sql expression in there and it automatically calculates. And do you plan to make the sqlite db scalable, or add the ability to include mysql for example. And will pocketbase have an official host?

Sorry for so many questions and suggestions, but I'm really excited about pocketbase, and I believe in your project with all my heart. Thanks again for the solved question.

**Reply by ganigeorgiev** [reply]
> And I thought it would be great to add a new type of columns to pocketbase, generated columns. You put a sql expression in there and it automatically calculates. 

See https://github.com/pocketbase/pocketbase/issues/4638.

> And will pocketbase have an official host?

No, I'm not planning offering hosting (or monetizing the project in any way). PocketBase is and will remain a small non-commercial hobby project and I don't intend growing it beyond what I can maintain on my own.

**Reply by lixelv** [reply]
Thanks for being open source, this is really cool!
